### PR TITLE
Add String method for functions

### PR DESCRIPTION
**Pull request by stevelinton**
A bit of a hack as it uses string streams rather than modifying the kernel

### CODE REVIEW

File: lib/function.gi [hunk 1]
```diff
@@ -59,3 +59,16 @@ function(func)
     Append(result, " ) ... end");
     return result;
 end);
+
+
+InstallMethod(String, "for a function, using string stream", [IsFunction] ,
```
Comment (fingolfin) [inline lib/function.gi]: remove space before comma?

File: lib/function.gi [hunk 2]
```diff
@@ -59,3 +59,16 @@ function(func)
     Append(result, " ) ... end");
     return result;
 end);
+
+
+InstallMethod(String, "for a function, using string stream", [IsFunction] , 
+        function(fun)
+    local  s, str;
+    s := "";
+    str := OutputTextString(s,true);
```
Comment (fingolfin) [inline lib/function.gi]: insert space after comma?

File: lib/function.gi [hunk 3]
```diff
@@ -59,3 +59,16 @@ function(func)
     Append(result, " ) ... end");
     return result;
 end);
+
+
+InstallMethod(String, "for a function, using string stream", [IsFunction] , 
+        function(fun)
```
Comment (fingolfin) [inline lib/function.gi]: Why this odd indentation?

**Comment by stevelinton** [comment]
@fingolfin I've addressed your comments. I've also made the original method a method for DisplayString. The String method now returns a version with  normalised whitespace, which is
ugly for large functions, but I think more appropriate for String.

File: lib/function.gi [hunk 4]
```diff
@@ -59,3 +59,26 @@ function(func)
...
+
+
+InstallMethod(DisplayString, "for a function, using string stream", [IsFunction], 
+function(fun)
+    local  s, str;
+    s := "";
+    str := OutputTextString(s, true);
+    PrintTo(str, fun);
+    CloseStream(str);
+    Add(s,'\n');
```
Comment (fingolfin) [inline lib/function.gi]: Space after comma?

File: lib/function.gi [hunk 5]
```diff
@@ -59,3 +59,26 @@ function(func)
...
 end);
+
+
+InstallMethod(DisplayString, "for a function, using string stream", [IsFunction], 
+function(fun)
+    local  s, str;
+    s := "";
+    str := OutputTextString(s, true);
+    PrintTo(str, fun);
+    CloseStream(str);
```
Comment (fingolfin) [inline lib/function.gi]: Super-nitpick, I am not expecting any action on it, but needed to get it off my chest: When I see "str" I think "string", but here it is "stream", while "s" is a string... This function is short, and so it's harmless, but I'd still rename at least one of them (probably `str` to `stream`, as I see strings more more often than streams, at least in GAP code)

**Comment by stevelinton** [comment]
Fixed nitpicks. Just wait for the tests to make sure I didn't break something.

**Pull request merged**